app/views/layouts/base.html.erb

```diff
@@ -18,7 +18,7 @@
     <%= csrf_meta_tags %>
 
     <script type="text/javascript">
-      var URL_PREFIX = '<%= request.script_name %>';
```

Comment:
`script_name` is probably not what we want.

Reply:
I'm not sure tbh, this seems to be how Rack gets the base URL - where as `controller.relative_url_root` reads the rails config, which I don't think we use. I think when installing on a sub-folder (e.g. example.com/foreman), we somehow set up the apache to pass that as a base. @ekohl perhaps you know how that is set?

Reply:
According to https://github.com/theforeman/puppet-foreman/blob/6b7d827cb515926a2b0b57febd9fd8a2aa8e43ef/templates/_suburi.conf.erb#L1 and https://github.com/theforeman/foreman/blob/a68ffcb24319e30c1724de40f1a445f77d86c319/config.ru#L9 and with respect to Rails documentation https://guides.rubyonrails.org/configuring.html#deploy-to-a-subdirectory-relative-url-root. It should work.
But dunno how to test it.

Reply:
Ok, seems like this makes sense then. TBH I don't know how many people actually install foreman in a subdir...

Reply:
I'd say some do if they want to try the foreman out, so they install it on some existing test server along side with other test apps, so I'd say we should keep it working, so the people don't need separate server to test foreman out, but I'd not consider it production setup ... :)

Reply:
Technically the installer has support for it by passing in `--foreman-url https://foreman.example.com/foreman`. I actually think it's a valuable thing to keep doing correctly and I've been pushing the folks to use the correct wrappers. It's probably broken in various places but being portable is IMHO a good thing. URL wrappers are a good thing (or hardcoding URLs/paths a bad thing).

Reply:
Well if I'm thinking about this correctly, if `controller.relative_url_root` wouldn't work, no paths would support subdirs, right? So it needs to work.

Reply:
Ok, it's not worth the argue I'd guess:
You can't even login in, if you use subdirectory because of: https://github.com/theforeman/foreman/blame/develop/webpack/assets/javascripts/react_app/components/LoginPage/helpers.js#L29

Reply:
FYI: if you press `y` in the GH UI it gives you the permanent URL. So https://github.com/theforeman/foreman/blame/a68ffcb24319e30c1724de40f1a445f77d86c319/webpack/assets/javascripts/react_app/components/LoginPage/helpers.js#L29

Reply:
So cool :heart_eyes: I'm always using the tripple dot, but it's not in the blame mode :disappointed: